libsledge/src/table_instructions.c

```diff
@@ -44,12 +55,22 @@ get_function_from_table(uint32_t idx, uint32_t type_id)
 {
 	assert(sledge_abi__current_wasm_module_instance.table != NULL);
 
-	assert(idx < sledge_abi__current_wasm_module_instance.table->capacity);
+	if (unlikely(idx >= sledge_abi__current_wasm_module_instance.table->capacity)) {
+		fprintf(stderr, "idx: %u, Table size: %u\n", idx, INDIRECT_TABLE_SIZE);
+		sledge_abi__wasm_trap_raise(WASM_TRAP_INVALID_INDEX);
+	}
 
 	struct sledge_abi__wasm_table_entry f = sledge_abi__current_wasm_module_instance.table->buffer[idx];
 
-	assert(f.type_id == type_id);
-	assert(f.func_pointer != NULL);
+	if (unlikely(f.type_id != type_id)) {
+		fprintf(stderr, "Function Type mismatch. Expected: %u, Actual: %u\n", type_id, f.type_id);
```

Comment:
Consider the fault information being put into a known location in memory that can be accessed when the longjmp returns.